ghost/external-media-inliner/lib/ExternalMediaInliner.js

```diff
@@ -63,21 +72,22 @@ class ExternalMediaInliner {
      * @param {Object} response - response from request
      * @returns {Object}
      */
-    #extractFileDataFromResponse(requestURL, response) {
-        const headers = response.headers;
-        const contentType = headers['content-type'];
+    async extractFileDataFromResponse(requestURL, response) {
+        const fileInfo = await FileType.fromBuffer(response.body);
+        const extension = fileInfo.ext;
 
-        const filename = requestURL
-            .split('/')
-            .pop()
-            .split('#')[0]
-            .split('?')[0];
+        const removeExtRegExp = new RegExp(`.${extension}`, '');
+        const fileNameNoExt = path.parse(requestURL).base.replace(removeExtRegExp, '');
 
-        const extension = mime.extension(contentType) || filename.split('.').pop();
+        // CASE: Query strings _can_ form part of the unique image URL, so rather that strip them include the in the file name, and then trim to 200 chars
+        // 200 is a magic number which could be refined. Also trim leading & trailing dashes.
```

Comment:
Let's at least use one of the real limiting numbers here, like for windows it's 255 chars according to [this](https://stackoverflow.com/a/265782)

Reply:
but doesn't really matter much  😄

Reply:
I landed on 248. Saw 255, 257, etc with 248 being the limit for folders. Airing on the side of caution makes sense to me here. The final file URL will be different anyway, so it isn't that important.

Reply:
Cool-cool 👍  nbd anyway, just wondered if we could put a boundary that has some meaning.